Framework/NexusGeometry/src/NexusGeometrySave.cpp

```diff
@@ -785,11 +786,25 @@ void saveInstrument(const Geometry::ComponentInfo &compInfo,
   NexusGeometrySave::saveNXSample(rootGroup, compInfo);
 
   // save NXdetectors
-  for (size_t index = compInfo.root() - 1; index > detInfo.size(); --index) {
+  std::list<size_t> nxDetectorCandidates;
+  for (size_t index = compInfo.root() - 1; index >= detInfo.size(); --index) {
     if (Geometry::ComponentInfoBankHelpers::isSaveableBank(compInfo, index)) {
+      nxDetectorCandidates.push_back(index);
+    }
+  }
+
+  for (std::list<size_t>::iterator index = nxDetectorCandidates.begin();
```

Comment:
Can you please combine these loops. 

1. Determine if potentially saveable, (passes `isSaveableBank`) 
2. Then check ancestors using `std::find` would be better.
3. Append to container once has been saved.

Can all be done in one loop.

Reply:
done